lsp-notifications.el

```diff
@@ -19,9 +19,21 @@
 (require 'cl-lib)
 (require 'subr-x)
 
+(defcustom lsp-inhibit-message nil
+  "If non-nil, inhibit the message echo via `inhibit-message'."
+  :type 'boolean
+  :group 'lsp-mode)
+
 (defun lsp--window-show-message (params)
-  (message "%s" (lsp--propertize (gethash "message" params)
-                                 (gethash "type" params))))
+  "Send the server's messages to message, inhibit if `lsp-inhibit-message'
+is set."
+  (let* ((prev-inhibit-message inhibit-message))
```

Comment:
Why not simply `let` bind `inhibit-message`?

```emacs-lisp
(let ((inhibit-message lsp-inhibit-message))
  (message "%s" (lsp--propertize (gethash "message" params)
				 (gethash "type" params))))
```

Reply:
Because I've been doing too much PHP and not enough Lisp and I totally forgot that was possible. Fixed now.